Src/IronPython.Modules/mmap.cs

```diff
@@ -595,6 +825,36 @@ public void resize(long newsize) {
                         throw PythonOps.TypeError("mmap can't resize a readonly or copy-on-write memory map.");
                     }
 
+                    if (_handle is not null
+                        && (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || RuntimeInformation.IsOSPlatform(OSPlatform.Linux))) {
+                        // resize on Posix platforms
+                        try {
+                            if (_handle.IsInvalid) {
+                                throw PythonOps.OSError(PythonErrorNumber.EBADF, "Bad file descriptor");
+                            }
+                            _view.Flush();
+                            _view.Dispose();
+                            _file.Dispose();
+
+                            // Resize the underlying file as needed.
+                            int fd = unchecked((int)_handle.DangerousGetHandle());
+                            PythonNT.ftruncateUnix(fd, newsize);
+
+    #if NET8_0_OR_GREATER
+                            _file = MemoryMappedFile.CreateFromFile(_handle, _mapName, newsize, _fileAccess, HandleInheritability.None, leaveOpen: true);
+    #else
+                            _sourceStream?.Dispose();
+                            _sourceStream = new FileStream(new SafeFileHandle((IntPtr)fd, ownsHandle: false), FileAccess.ReadWrite);
+                            _file = CreateFromFile(_sourceStream, _mapName, newsize, _fileAccess, HandleInheritability.None, leaveOpen: true);
+    #endif
+                            _view = _file.CreateViewAccessor(_offset, newsize, _fileAccess);
+                            return;
+                        } catch {
+                            close();
```

Comment:
I think this `close` might be problematic (it won't release resources since `_refCount` is `2` at this point). Anyway, the Windows version has the same issue which I still need to figure out for https://github.com/IronLanguages/ironpython3/pull/1866 so we can leave it as-is for now. (Unless the "proper" solution is obvious to you and you feel like taking it on).

Reply:
Yes, it is problematic. I didn't look into the solutions, but the way `close` and `CloseWorker` work perplexes me. If the code calls `close` while `_refCount` is > 1, then `mmap` gets marked as closed but its fields are never properly disposed? That can't be right.

And yes, I'd rather leave it as it is, since it is not a regression, and I don't want to make this PR too much about `mmap`. It was supposed to be about `PusixFileStream` w/o regressions. As I was reading through the `mmap` code and trying to figure out the rules of engagement and lifietime management of various things, I have noticed a few other suspicious pre-existing pieces, which I will need to examine more carefully, and PR if needed.